packages/controls/src/lib/add-index-signature.ts

```diff
@@ -0,0 +1,5 @@
+// See https://github.com/microsoft/TypeScript/issues/15300
+export type AddIndexSignature<T> =
```

Comment:
This is a legalized version of https://github.com/makeswift/makeswift/pull/781/files#diff-7135d2d9b277af3c39d03ead4f2e0d109d50c72190c9f8ed5ae415d410afdd32. 

The final comment from the TS team on this issue is "[interfaces should declare an explicit index signature if they intend to be indexed](https://github.com/microsoft/TypeScript/issues/15300#issuecomment-1320620897)."